exercises/house/example.go

```diff
@@ -1,39 +1,69 @@
 package house
 
-func Embed(relPhrase, nounPhrase string) string {
-	return relPhrase + " " + nounPhrase
+const testVersion = 2
+
+var songLines = []string{
+	"the horse and the hound and the horn\nthat belonged to",
+	"the farmer sowing his corn\nthat kept",
+	"the rooster that crowed in the morn\nthat woke",
+	"the priest all shaven and shorn\nthat married",
+	"the man all tattered and torn\nthat kissed",
+	"the maiden all forlorn\nthat milked",
+	"the cow with the crumpled horn\nthat tossed",
+	"the dog\nthat worried",
+	"the cat\nthat killed",
+	"the rat\nthat ate",
+	"the malt\nthat lay in",
 }
 
-func Verse(subject string, relPhrases []string, nounPhrase string) string {
-	return subject + " " + recurse(relPhrases, nounPhrase)
+// Recursive Solution
+
+func Verse(v int) (verse string) {
+	v--
+	i := len(songLines) - v
+	verse += buildVerse(songLines[i:], "This is ")
+	verse += "the house that Jack built."
+	return
 }
 
-func recurse(relPhrases []string, nounPhrase string) string {
-	if len(relPhrases) == 0 {
-		return nounPhrase
+func buildVerse(songLines []string, cur string) string {
+	if len(songLines) == 0 {
+		return cur
 	}
-	return Embed(relPhrases[0], recurse(relPhrases[1:], nounPhrase))
+	cur += songLines[0]
+	cur += " "
+	return buildVerse(songLines[1:], cur)
 }
 
-func Song() string {
-	relPhrases := []string{
-		"the horse and the hound and the horn\nthat belonged to",
-		"the farmer sowing his corn\nthat kept",
-		"the rooster that crowed in the morn\nthat woke",
-		"the priest all shaven and shorn\nthat married",
-		"the man all tattered and torn\nthat kissed",
-		"the maiden all forlorn\nthat milked",
-		"the cow with the crumpled horn\nthat tossed",
-		"the dog\nthat worried",
-		"the cat\nthat killed",
-		"the rat\nthat ate",
-		"the malt\nthat lay in",
-	}
-	subject := "This is"
-	nounPhrase := "the house that Jack built."
-	s := subject + " " + nounPhrase
-	for c := len(relPhrases) - 1; c >= 0; c-- {
-		s += "\n\n" + Verse(subject, relPhrases[c:], nounPhrase)
+func Song() (song string) {
+	for i := 0; i <= len(songLines); i++ {
+		song += Verse(i + 1)
+		if i < len(songLines) {
+			song += "\n\n"
+		}
 	}
-	return s
+	return
 }
+
+// Iterative Solution
```

Comment:
I see the primary purpose of the example as making sure that the tests are in fact solvable, by being used in Travis CI.

If we consider only this purpose, a commented-out example serves no benefit!

Though that's not the whole story, since I do see a secondary purpose: To be (as its name suggests) an example of a working solution, available for students to look at. But they have to work a bit hard to find it, since they have to know about the existence of this git repo and examine this file.

The commented-out example is, of course, useful for this purpose.

I don't feel strongly either way; it surely can be kept.

If there is great interest in actually putting multiple examples though Travis CI (this will help us assure they are good examples for students to look at), that is a separate project.

Haskell has done this at https://github.com/exercism/xhaskell/pull/396 ; it was important for that track for slightly different reasons, since the Haskell track wished to allow its students to submit functions with various type signatures, so the multiple examples help ensure that multiple type signatures are in fact accepted by the tests.